controllers/admin_client.go

```diff
@@ -216,9 +216,18 @@ func (client *MockAdminClient) GetStatus() (*fdbtypes.FoundationDBStatus, error)
 			continue
 		}
 
-		for processIndex := 1; processIndex <= processCount; processIndex++ {
-			fullAddress := client.Cluster.GetFullAddress(pod.Status.PodIP, processIndex)
+		subs, err := podClient.GetVariableSubstitutions()
+		if err != nil {
+			return nil, err
+		}
 
+		podIP, ok := subs["FDB_PUBLIC_IP"]
+		if !ok {
+			podIP = pod.Status.PodIP
+		}
+
+		for processIndex := 1; processIndex <= processCount; processIndex++ {
+			fullAddress := client.Cluster.GetFullAddress(podIP, processIndex)
```

Comment:
FDB will set the first address (or the TLS address) provided via `public_address` the previous behaviour was actually wrong we just never noticed it.